src/main/java/core/screens/HUD/HudManager.java

```diff
@@ -0,0 +1,108 @@
+package core.screens.HUD;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.OrthographicCamera;
+import com.badlogic.gdx.graphics.g2d.BitmapFont;
+import com.badlogic.gdx.graphics.g2d.SpriteBatch;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.utils.Disposable;
+import com.badlogic.gdx.utils.viewport.FitViewport;
+import com.badlogic.gdx.utils.viewport.Viewport;
+
+public class HudManager implements Disposable {
+
+    public Stage stage;
+    private Viewport viewport;
+
+    //score && time tracking variables
+    private Integer worldTimer;
+    private float timeCount;
+    private static Integer score;
+    private boolean timeUp;
+
+    //Scene2D Widgets
+    private Label countdownLabel, timeLabel, linkLabel;
+    private static Label scoreLabel;
+
+    public HudManager(SpriteBatch sb) {
+        //define tracking variables
+        worldTimer = 250;
+        timeCount = 0;
+        score = 0;
+
+        //setup the HUD viewport using a new camera seperate from gamecam
+        //define stage using that viewport and games spritebatch
+        viewport = new FitViewport(Gdx.graphics.getWidth(), Gdx.graphics.getHeight(), new OrthographicCamera());
+        stage = new Stage(viewport, sb);
+
+        //define labels using the String, and a Label style consisting of a font and color
+        countdownLabel = new Label(String.format("%03d", worldTimer), new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+        scoreLabel = new Label(String.format("%06d", score), new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+        timeLabel = new Label("LEFTOVER TIME", new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+        linkLabel = new Label("POINTS", new Label.LabelStyle(new BitmapFont(), Color.WHITE));
+
+        //define a table used to organize hud's labels
+        Table table = new Table();
+        table.top();
+        table.setFillParent(true);
+
+        //add labels to table, padding the top, and giving them all equal width with expandX
+        table.add(linkLabel).expandX().padTop(10);
+        table.add(timeLabel).expandX().padTop(10);
+        table.row();
+        table.add(scoreLabel).expandX();
+        table.add(countdownLabel).expandX();
```

Comment:
a lot going on here, move to own method or separate class if renaming HUDManager -> GameOverlayManager